src/tree_schema.c

```diff
@@ -184,6 +184,73 @@ lys_getnext(const struct lysc_node *last, const struct lysc_node *parent, const
     return next;
 }
 
+API const struct lysc_node *
+lys_find_node(struct ly_ctx *ctx, const struct lysc_node *context_node, const char *qpath)
+{
+    const char *id = qpath;
+    const char *prefix, *name;
+    size_t prefix_len, name_len;
+    unsigned int u;
+    const struct lysc_node *node = context_node;
+    struct lys_module *mod = NULL;
+
+    LY_CHECK_ARG_RET(ctx, qpath, NULL);
+
+    while(*id) {
+        if (id[0] == '/') {
+            ++id;
+        }
+        /* precess ".." in relative paths */
+        while (!strncmp("../", id, 3)) {
+            id += 3;
+            if (!node) {
+                LOGERR(ctx, LY_EINVAL, "Invalid qpath \"%s\" - too many \"..\" in the path.", qpath);
+                return NULL;
+            }
+            node = node->parent;
+        }
```

Comment:
I wondered whether `./` should be parsed in this function?

Reply:
no, I'll add it

Reply:
well, the question is if it should be supported. There is no path definition in the YANG ABNF grammar (RFC 7950) that would support `./`, so at the end I would prefer to avoid supporting it